src/helper.js

```diff
@@ -537,41 +542,37 @@ class helper {
 
 		let mosaicsFieldObject = [];
 
-		for (const resolvedMosaic of resolvedMosaics) {
-			if (resolvedMosaic.id.toHex() === http.networkCurrency.mosaicId) {
-				mosaicsFieldObject.push(this.networkCurrencyMosaicBuilder(resolvedMosaic));
-			} else {
-				if (0 < mosaicInfos.length && 0 < mosaicNames.length) {
-					let { divisibility } = mosaicInfos.find(info => info.mosaicId === resolvedMosaic.id.toHex());
-
-					mosaicsFieldObject.push({
-						...resolvedMosaic,
-						rawAmount: resolvedMosaic.amount,
-						mosaicId: resolvedMosaic.id.toHex(),
-						amount: helper.formatMosaicAmountWithDivisibility(resolvedMosaic.amount, divisibility),
-						mosaicAliasName: MosaicService.extractMosaicNamespace({ mosaicId: resolvedMosaic.id.toHex() }, mosaicNames)
-					});
-				}
-			}
-		}
+		const uniqueMosaicIds = [...new Set(resolvedMosaics.map(mosaic => mosaic.id.toHex()))];
 
-		return mosaicsFieldObject;
-	}
+		uniqueMosaicIds.forEach(idHex => {
+			const mosaics = resolvedMosaics.filter(mosaic => mosaic.id.toHex() === idHex);
 
-	static networkCurrencyMosaicBuilder = mosaic => {
-		if (!(mosaic instanceof Mosaic))
-			throw new Error('It required Mosaic instance.');
+			const sumAmount = mosaics.reduce((acc, cur) => acc + Number(cur.amount.toString()), 0);
 
-		if (mosaic.id.toHex() !== http.networkCurrency.mosaicId)
-			throw new Error('Mosaic id does not match network Currency.');
+			const mosaicField = {
+				...mosaics[0],
+				rawAmount: UInt64.fromUint(sumAmount),
+				mosaicId: mosaics[0].id.toHex()
+			};
 
-		return {
-			...mosaic,
-			rawAmount: mosaic.amount,
-			mosaicId: mosaic.id.toHex(),
-			amount: this.formatMosaicAmountWithDivisibility(mosaic.amount, http.networkCurrency.divisibility),
-			mosaicAliasName: http.networkCurrency.namespaceName
-		};
+			if (idHex === http.networkCurrency.mosaicId) {
+				Object.assign(mosaicField, {
+					amount: this.formatMosaicAmountWithDivisibility(sumAmount, http.networkCurrency.divisibility),
+					mosaicAliasName: http.networkCurrency.namespaceName
+				});
+			} else {
+				const { divisibility } = mosaicInfos.find(info => info.mosaicId === mosaics[0].id.toHex());
+
+				Object.assign(mosaicField, {
+					amount: this.formatMosaicAmountWithDivisibility(sumAmount, divisibility),
+					mosaicAliasName: MosaicService.extractMosaicNamespace({ mosaicId: mosaics[0].id.toHex() }, mosaicNames)
+				});
+			}
+
+			mosaicsFieldObject.push(mosaicField);
+		});
+
+		return mosaicsFieldObject;
```

Comment:
i think you can use map instead of forEach?

Reply:
That makes sense, I can just return rather than build and push to array.